File: Day 2/Day 2.py
# Advent of Code Day 2

# Part 1
passwords = [line.rstrip('\n') for line in open("passwords.txt")]
x = 0

for i in range(len(passwords)):
    numbers = passwords[i].split()[0].split("-")
    char = passwords[i].split()[1].strip(":")
    pass_split = passwords[i].split()[2]
    minNum = int(numbers[0])
    maxNum = int(numbers[1])
    if pass_split.count(char) >= minNum and pass_split.count(char) <= maxNum:
        print(pass_split)
        x += 1
print("The are", x, "valid passwords")

# Part 2
passwords = [line.rstrip('\n') for line in open("passwords.txt")]
x = 0

for i in range(len(passwords)):
    numbers = passwords[i].split()[0].split("-") 
    char = passwords[i].split()[1].strip(":")
    pass_split = passwords[i].split()[2]
    firstNum = int(numbers[0]) - 1
    secondNum = int(numbers[1]) - 1
    if pass_split[firstNum] != pass_split[secondNum] and (
            pass_split[firstNum] == char or pass_split[secondNum] == char):
        x += 1

print("The are", x, "valid passwords")

File: Day 2/Day 2.php
<!-- # Advent of Code Day 2 -->

<?php
$part1 = 0;
$part2 = 0;
foreach (explode("\n", file_get_contents("passwords.txt")) as $pass) {
$pass = explode(" ", $pass);
$policy = explode("-", $pass[0]);
foreach (count_chars($pass[2], 1) as $i => $val) {
if (str_replace(":", "", $pass[1]) == chr($i) && $policy[0] <= $val && $policy[1] >= $val) {
$part1++;
        }
    }
if (str_replace(":", "", $pass[1]) === $pass[2][$policy[0] - 1] || str_replace(":", "", $pass[1]) === $pass[2][$policy[1] - 1]) {
if ($pass[2][$policy[0] - 1] !== $pass[2][$policy[1] - 1]){
$part2++;
        } 
    }
} 
echo "Part 1:<br>" . $part1 . "<br><br>Part 2:<br>" . $part2; 
?>